channeldb/mp_payment_test.go

```diff
@@ -368,6 +368,181 @@ func TestNeedWaitAttempts(t *testing.T) {
 	}
 }
 
+// TestAllowMoreAttempts checks whether more attempts can be created against
+// ALL possible payment statuses.
+func TestAllowMoreAttempts(t *testing.T) {
```

Comment:
yeah I'd love to, but don't think there's an easy way to implement it, similar [comments here](https://github.com/lightningnetwork/lnd/pull/7391/files#r1320670644). We could do a similar setup as `resultCollector`, but I feel it's a bit too much for this test.

Reply:
Which ones are contradicting? Like this one,
```
{
			// With zero remainingAmt and both settled HTLCs and
			// failed payment, we don't allow more HTLC attempts.
			status:         StatusInFlight,
			remainingAmt:   0,
			hasSettledHTLC: true,
			paymentFailed:  true,
			allowMore:      false,
			expectedErr:    nil,
		},
```
Because we have in-flight HTLCs, this would not be `StatusSucceeded` or `StatusFailed`, and it's a possible state from `decidePaymentStatus`.

Reply:
Cool updated the tests. So the cases here are made for `AllowMoreAttempts` only, how they are made is by analyzing the logical branches in that method and make sure as many branches are cover as possible. This is also why this is unit test, in that you don't care much context outside this `AllowMoreAttempts` method, and these variables are just state bits that turn on/off certain branches.